source/cellular_bg96_api.c

```diff
@@ -1700,21 +1780,39 @@ static CellularPktStatus_t _Cellular_RecvFuncGetPsmSettings( CellularContext_t *
 }
 
 /*-----------------------------------------------------------*/
+static CellularPktStatus_t socketRecvDataPrefixSsl( void * pCallbackContext,
+                                                 char * pLine,
+                                                 uint32_t lineLength,
+                                                 char ** ppDataStart,
+                                                 uint32_t * pDataLength )
+{
+    return socketRecvDataPrefix( pCallbackContext, pLine, lineLength, ppDataStart, pDataLength, true );
+}
 
-static CellularPktStatus_t socketRecvDataPrefix( void * pCallbackContext,
+static CellularPktStatus_t socketRecvDataPrefixNonSsl( void * pCallbackContext,
                                                  char * pLine,
                                                  uint32_t lineLength,
                                                  char ** ppDataStart,
                                                  uint32_t * pDataLength )
+{
+    return socketRecvDataPrefix( pCallbackContext, pLine, lineLength, ppDataStart, pDataLength, false);
+}
+static CellularPktStatus_t socketRecvDataPrefix( void * pCallbackContext,
+                                                 char * pLine,
+                                                 uint32_t lineLength,
+                                                 char ** ppDataStart,
+                                                 uint32_t * pDataLength,
+                                                 bool useSsl )
 {
     char * pDataStart = NULL;
     uint32_t prefixLineLength = 0U;
     int32_t tempValue = 0;
     CellularATError_t atResult = CELLULAR_AT_SUCCESS;
     CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
     uint32_t i = 0;
-    char pLocalLine[ MAX_QIRD_STRING_PREFIX_STRING + 1 ] = "\0";
-    uint32_t localLineLength = MAX_QIRD_STRING_PREFIX_STRING > lineLength ? lineLength : MAX_QIRD_STRING_PREFIX_STRING;
+    char pLocalLine[ (useSsl ? MAX_QSSLRECV_STRING_PREFIX_STRING : MAX_QIRD_STRING_PREFIX_STRING) + 1 ];
+    strcpy(pLocalLine, "\0"); 
+    uint32_t localLineLength = (useSsl ? MAX_QSSLRECV_STRING_PREFIX_STRING : MAX_QIRD_STRING_PREFIX_STRING) > lineLength ? lineLength : (useSsl ? MAX_QSSLRECV_STRING_PREFIX_STRING : MAX_QIRD_STRING_PREFIX_STRING);

```

Comment:
this expression `useSsl ? MAX_QSSLRECV_STRING_PREFIX_STRING : MAX_QIRD_STRING_PREFIX_STRING` is repeated a lot of times, put it in a variable

Reply:
Changed